autodiff/forward/eigen.hpp

```diff
@@ -54,6 +54,23 @@ template<> struct NumTraits<autodiff::dual> : NumTraits<double> // permits to ge
     };
 };
 
+template<> struct NumTraits<autodiff::HigherOrderDual<2>> : NumTraits<double> // permits to get the epsilon, dummy_precision, lowest, highest functions
```

Comment:
Good to see you back @supersega ! :)
Is it possible to make this generic with respect to the order (instead of 2, a `std::size_t N`)? Not sure if necessary though, if we just want to use eigen up to 2nd order to compute Hessian matrices.

Reply:
Hi @allanleal same to you. Probably yes, but I'm not sure. We can improve code quality in nex PR's, since this one is too big now (IMHO).

Reply:
Hi @allanleal , sorry for the delay(again), I've tied to avoid duplication of code using templates, but it looks like we can use the only macro here. So, let merge this PR, and then make new things based on it =)